app/src/main/java/com/onboarding/calculator/mvp/presenter/CalculatorPresenter.java

```diff
@@ -20,4 +18,31 @@ public void onCalculatorButtonPressed(String buttonText) {
         view.showValues(model.getLastModified());
     }
 
+    @Override
+    public void onEqualsButtonPressed() {
+        Double result = model.getResult();
+        if (result != null) {
+            view.showValues(result.toString());
+        } else {
+            view.showError();
+        }
+
+    }
+
+    @Override
+    public boolean deleteAll() {
+        model.reset();
+        view.showValues(model.getLastModified());
+        view.showDeleteAllMessage();
+        return true;
```

Comment:
why do you need this boolean return? you are not using it on the activity

Reply:
Solved!